src/test/java/edu/harvard/iq/dataverse/validation/PasswordValidatorTest.java

```diff
@@ -28,203 +33,102 @@
+            // add a fourth characteristic
+            Arguments.of(3, "7 Potato",  goodStrength20, maxLength, minLength, dictionary, 4, characterRulesHarvardLevel3, numConsecutiveDigitsAllowed),
+            // Now it does not matter: 20 characters
+            Arguments.of(0, "7 Potato901234567890", goodStrength20, maxLength, minLength, dictionary, 4, characterRulesHarvardLevel3, numConsecutiveDigitsAllowed),
+            // Now we use all four
+            Arguments.of(0, "8.Potato",  goodStrength20, maxLength, minLength, dictionary, 4, characterRulesHarvardLevel3, numConsecutiveDigitsAllowed),
+            Arguments.of(2, "Potato.Too.12345.Short", 0, maxLength, 23, dictionary, numberOfCharacters, characterRulesHarvardLevel3, numConsecutiveDigitsAllowed),
+            Arguments.of(0, "Potatoes on my plate with beef", 30, maxLength, minLength, dictionary, numberOfCharacters, characterRulesHarvardLevel3, numConsecutiveDigitsAllowed),
+            Arguments.of(0, "Potatoes on my plate with pie.", 30, maxLength, minLength, dictionary, numberOfCharacters, characterRulesHarvardLevel3, numConsecutiveDigitsAllowed),
+            Arguments.of(0, "Potatoes on a plate  .",  30, maxLength, minLength, dictionary, numberOfCharacters, characterRulesHarvardLevel3, numConsecutiveDigitsAllowed),
+            // Pass when repeating character maximum is 5
+            Arguments.of(0, "Repeated Potatoes:0000",  30, maxLength, minLength, dictionary, numberOfCharacters, characterRulesHarvardLevel3, 5),
+            // Allow no more than 3 repeating characters (default)
+            Arguments.of(0, "Repeated Potatoes:000",  30, maxLength, minLength, dictionary, numberOfCharacters, characterRulesHarvardLevel3, numConsecutiveDigitsAllowed),
+            //For some reason, whitespace doesn't count in the repeating rule?
+            Arguments.of(6, "          ", 30, maxLength, minLength, dictionary, numberOfCharacters, characterRulesHarvardLevel3, numConsecutiveDigitsAllowed),
+            // Good enough for Dataverse 4.0.
+            Arguments.of(0, "potat1", goodStrength20, maxLength, 6, dictionary, numberOfCharactersDefault, characterRulesDefault, numConsecutiveDigitsAllowed),
+            // Has repeating chars exceeding limit, but goodstrength waives it
+            Arguments.of(0, "potat000000000000000", goodStrength20, maxLength, 6, dictionary, numberOfCharactersDefault, characterRulesDefault, numConsecutiveDigitsAllowed),
+            // 5 or more numbers in a row
+            Arguments.of(2, "ma02138", goodStrength20, maxLength, 6, dictionary, numberOfCharactersDefault, characterRulesDefault, numConsecutiveDigitsAllowed),
+            // 5 or more numbers in a row, but multiple times
+            Arguments.of(3, "ma8312002138", goodStrength20, maxLength, 6, dictionary, numberOfCharactersDefault, characterRulesDefault, numConsecutiveDigitsAllowed)
```

Comment:
Is it understood why this has 3 errors instead of 2 with the new version of passay?

Reply:
Yes. There was a bug in the older Passay version, which wasn't identifying multiple occurences of more than x numbers in a row. (It's 10 digits here, which is 2x error of 5 numbers) This made a +1 for this test necessary.